eng/references.targets

```diff
@@ -1,33 +1,43 @@
 <Project>
   <PropertyGroup>
-    <ContractDependencyPaths>$(RefPath)</ContractDependencyPaths>
+    <IncludeDefaultReferences Condition="'$(IncludeDefaultReferences)' == ''">true</IncludeDefaultReferences>
+    <IncludeNETStandardReference Condition="'$(IncludeDefaultReferences)' == 'true' and
+                                            '$(TargetFrameworkIdentifier)' != '.NETStandard' and
+                                            '$(IsReferenceAssembly)' != 'true'">true</IncludeNETStandardReference>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(DisableImplicitFrameworkReferences)' == 'true'">
+    <AssemblySearchPaths>$(AssemblySearchPaths);$(RefPath)</AssemblySearchPaths>
     <FrameworkPathOverride>$(RefPath)</FrameworkPathOverride>
-    <AssemblySearchPaths>$(AssemblySearchPaths);$(RefPath);{RawFileName}</AssemblySearchPaths>
     <!-- Disable RAR from transitively discovering dependencies for References -->
     <_FindDependencies>false</_FindDependencies>
-    <!--
-      We don't use any of MSBuild's resolution logic for resolving the framework, so just set these two properties to any folder
-      that exists to skip the GenerateReferenceAssemblyPaths task (not target) and to prevent it from outputting a warning (MSB3644).
-      Need to set these after the common targets import.
-      -->
-    <_TargetFrameworkDirectories>$(MSBuildThisFileDirectory)</_TargetFrameworkDirectories>
-    <_FullFrameworkReferenceAssemblyPaths>$(MSBuildThisFileDirectory)</_FullFrameworkReferenceAssemblyPaths>
-  </PropertyGroup>
-  <PropertyGroup Condition="'$(IncludeDefaultReferences)' == ''">
-    <IncludeDefaultReferences Condition="'$(MSBuildProjectExtension)' == '.csproj'">true</IncludeDefaultReferences>
-    <IncludeDefaultReferences Condition="'$(MSBuildProjectExtension)' == '.vbproj'">true</IncludeDefaultReferences>
   </PropertyGroup>
 
-  <Target Name="SetupDefaultReferences">
-    <ItemGroup Condition="'$(IncludeDefaultReferences)' == 'true'">
-      <!-- netstandard is a default reference whenever building for NETStandard or building an implementation assembly -->
-      <DefaultReference Condition="('$(TargetFrameworkIdentifier)' == '.NETStandard' or '$(IsReferenceAssembly)' != 'true') and Exists('$(RefPath)netstandard.dll')" Include="netstandard" />
+  <Target Name="UpdateReferenceItems"
+          BeforeTargets="PrepareForBuild">
+    <!-- Reference everything in the targeting pack directory. Don't reference the same contract assembly or experimental assemblies. -->
+    <ItemGroup Condition="'$(IsTestProject)' == 'true' or '$(IsTestSupportProject)' == 'true'">
+      <Reference Include="$(RefPath)*.dll"
+                 Exclude="$(RefPath)$(TargetFileName)"
+                 Private="false" />
+      <!-- Filter out experimental reference assemblies. -->
+      <Reference Remove="@(Reference)"
+                 Condition="$([System.String]::new('%(FileName)%(Extension)').Contains('Experimental'))" />
```

Comment:
Has this been like that always?

Reply:
For Utf8String yes: https://github.com/dotnet/runtime/blob/e0e752ac9ee67f3dc7fa127d6f7495132b4bcee7/src/libraries/Directory.Build.props#L358-L365.